src/Sentry.Protocol/CollectionsExtensions.cs

```diff
@@ -9,7 +9,7 @@ internal static class CollectionsExtensions
             this ConcurrentDictionary<string, object> dictionary,
             string key)
             where TValue : class, new()
-            => dictionary.GetOrAdd(key, _ => new TValue()) as TValue;
+            => (TValue) dictionary.GetOrAdd(key, _ => new TValue());
```

Comment:
This is an important change @bruno-garcia. Previously it was using `as` which could return `null` if the value has been set before, but to a different type. I glanced through code and it seems that _shouldn't_ happen so I replaced it with an explicit conversion which would throw instead. Can you validate my assumptions here?